GiniVision/Classes/Core/Screens/Digital Invoice/GiniTextField.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  GiniTextField.swift
+//  GiniVision
+//
+//  Created by Maciej Trybilo on 18.12.19.
+//
+
+import UIKit
+
+protocol GiniTextFieldDelegate: class {
+    
+    func textDidChange(_ giniTextField: GiniTextField)
+}
+
+class GiniTextField: UIView {
+
+    private let stackView = UIStackView()
+    private let titleLabel = UILabel()
+    private let textFieldContainer = UIStackView()
+    private let prefixLabel = UILabel()
+    private let separatorView = UILabel()
+    private let textField = UITextField()
+    private let underscoreView = UIView()
+    
+    weak var delegate: GiniTextFieldDelegate?
+    
+    var title: String? {
+        didSet {
+            titleLabel.text = title
+        }
+    }
+    
+    var titleFont: UIFont {
+        
+        set {
+            titleLabel.font = newValue
+        }
+        
+        get {
+            return titleLabel.font
+        }
+    }
+    
+    var titleTextColor: UIColor {
+        
+        set {
+            titleLabel.textColor = newValue
+        }
+        
+        get {
+            return titleLabel.textColor
+        }
+    }
+    
+    var text: String? {
+        
+        set {
+            textField.text = newValue
+        }
+        
+        get {
+            return textField.text
+        }
+    }
+    
+    var prefixText: String? {
+        
+        didSet {
+            
+            if prefixText != nil {
+                
+                prefixLabel.isHidden = false
+                separatorView.isHidden = false
+                
+            } else {
+                prefixLabel.isHidden = true
+                separatorView.isHidden = true
+            }
+            
+            prefixLabel.text = prefixText
+        }
+    }
+    
+    var textFont: UIFont {
+        
+        set {
+            textField.font = newValue
+            prefixLabel.font = newValue
+        }
+        
+        get {
+            return prefixLabel.font
+        }
+    }
+    
+    var textColor: UIColor {
+        
+        set {
+            textField.textColor = newValue
+            prefixLabel.textColor = newValue
+        }
+        
+        get {
+            return prefixLabel.textColor
+        }
+    }
+    
+    var textFieldFirstBaselineAnchor: NSLayoutYAxisAnchor {
+        return textField.firstBaselineAnchor
+    }
+    
+    var textFieldLastBaselineAnchor: NSLayoutYAxisAnchor {
+        return textField.lastBaselineAnchor
+    }
+    
+    var keyboardType: UIKeyboardType {
+        
+        set {
+            textField.keyboardType = newValue
```

Comment:
I'm kind of indecisive about all these fields that essentially just delegate to a private member like `textField` here... At some point, there's not much difference between just having it public and this approach? But I've also nothing against it - just a thought rather than anything to fix.

Reply:
I thought about it as well. It might change as the type evolves, but at this point I have a slight preference towards tightly controlling the API. For instance `textFont` and `textColor` tie the `textField` and the `prefixLabel` together  and I think it simplifies things for the client code.